packages/compass-query-bar/src/components/query-bar.tsx

```diff
@@ -173,6 +197,33 @@ export const QueryBar: React.FunctionComponent<QueryBarProps> = ({
             placeholder={placeholders?.filter}
           />
         </div>
+        {showExplainButton && newExplainPlan && (
+          <>
+            <div ref={showExplainButtonRef}>
+              <Button
+                aria-label="Reset query"
+                data-testid="query-bar-reset-filter-button"
+                onClick={onExplain}
+                disabled={!isQueryValid}
+                size="small"
+                type="button"
+              >
+                Explain
+              </Button>
+            </div>
+            <GuideCue
```

Comment:
Should we hide the GuideCue once the button is clicked?

Reply:
Oh, you mean the explain button itself, right? Yeah, makes sense!

Reply:
Yes, hide the cue once explain button is clicked. We are doing same in aggregation guide-cues.

Reply:
Done in https://github.com/mongodb-js/compass/pull/4465/commits/72911f9127c8e9911bedfafdc1b5b612e199e612